Code/C++/Doubly_linkedlist/Doubly_linkedlist_all_operations.cpp

```diff
@@ -0,0 +1,390 @@
+    {
+        cout << "List is empty " << endl;
+    }
+    else
+    {
+        cout << "\ntraversing from left from right ";
+        while (ptr != nullptr)
+        {
+            cout << "\nInfo of node at index " << count << "--> " << ptr->data;
+            ptr = ptr->next;
+            count += 1;
+        }
+        count -= 1;
+        cout << "\nTransversing from right to left ";
+        while (p != nullptr)
+        {
+            cout << "\nInfo of node at index " << count << "--> " << p->data;
+            p = p->prev;
+            count -= 1;
+        }
+    }
+}
+
+//Function for inserting a node in beginning.
+
+void Doubly_linklist::insertion_at_beginning()
+{
+    node *new_node = new node;
+    cout << "\nEnter the info of the node ";
+    cin >> new_node->data;
+    if (start == nullptr && last == nullptr)
+    {
+        new_node->prev = nullptr;
+        new_node->next = nullptr;
+        start = last = new_node;
+    }
+    else
+    {
+        new_node->prev = nullptr;
+        new_node->next = start;
+        start->prev = new_node;
+        start = new_node;
+    }
+}
+
+//Function for inserting new node at end of list.
+
+void Doubly_linklist::insertion_at_end()
+{
+    node *new_node = new node;
+    cout << "\nEnter the info of node ";
+    cin >> new_node->data;
+    if (last == nullptr)
+    {
+        cout << "\nList is empty ";
+    }
+    else
+    {
+        new_node->next = nullptr;
+        last->next = new_node;
+        new_node->prev = last;
+        last = new_node;
+    }
+}
+
+// Function for inserting new node next to given position(index) in list.
+
+void Doubly_linklist::insertion_after_given_index()
+{
+    node *ptr = start;
+    node *new_node = new node;
+    cout << "\nEnter the info of the node you want to insert";
+    cin >> new_node->data;
+    int count = 0, index;
+    cout << "\nEnter the location where you want to insert new node";
+    cin >> index;
+    if (index < count)
+    {
+        cout << "\nInvalid position ";
+    }
+    else if (start == nullptr || last == nullptr)
+    {
+        cout << "\nList is empty ";
+    }
+    else if (index == 4)
+    {
+        insertion_at_end();
+    }
+    else
+    {
+        while (count != index)
+        {
+            ptr = ptr->next;
+            count += 1;
+        }
+        new_node->next = ptr->next;
+        ptr->next->prev = new_node;
+        ptr->next = new_node;
+        new_node->prev = ptr;
+    }
+}
+
+//Function to insert a node at given index.
+
+void Doubly_linklist::insertion_at_index()
+{
+    node *new_node = new node;
+    node *ptr = start;
+    cout << "\nEnter the info of the node you want to insert";
+    cin >> new_node->data;
+    int count = 0, index;
+    cout << "\nEnter the location where you want to insert new node";
+    cin >> index;
+    if (index < count)
+    {
+        cout << "\nInvalid position ";
+    }
+    else if (start == nullptr || last == nullptr)
+    {
+        cout << "\nList is empty ";
```

Comment:
In this case, if the index is 0, then the addition is possible just like creating the first node.